src/gui/accountsettings.cpp

```diff
@@ -243,82 +243,116 @@ void AccountSettings::slotCustomContextMenuRequested(const QPoint &pos)
         return;
-    } else if (!_model->data(index, FolderStatusDelegate::IsReady).toBool()) {
+    }
+
+    auto classification = _model->classify(index);
+    if (classification != FolderStatusModel::RootFolder && classification != FolderStatusModel::SubFolder) {
+        return;
+    }
+
+
+    // Only allow removal if the item isn't in "ready" state.
+    if (classification == FolderStatusModel::RootFolder && !_model->data(index, FolderStatusDelegate::IsReady).toBool()) {
         QMenu *menu = new QMenu(tv);
         menu->setAttribute(Qt::WA_DeleteOnClose);
         removeFolderAction(menu);
         menu->popup(QCursor::pos());
         return;
     }
 
-    if (_model->classify(index) == FolderStatusModel::SubFolder) {
-        QMenu *menu = new QMenu(tv);
-        menu->setAttribute(Qt::WA_DeleteOnClose);
+    QMenu *menu = new QMenu(tv);
+    menu->setAttribute(Qt::WA_DeleteOnClose);
 
-        QAction *ac = menu->addAction(tr("Open local folder"));
-        connect(ac, &QAction::triggered, this, &AccountSettings::slotOpenCurrentLocalSubFolder);
+    // Add an action to open the folder in the system's file browser:
 
+    QUrl folderUrl;
+    if (classification == FolderStatusModel::SubFolder) {
         QString fileName = _model->data(index, FolderStatusDelegate::FolderPathRole).toString();
-        if (!QFile::exists(fileName)) {
+        folderUrl = QUrl::fromLocalFile(fileName);
+    } else {
+        // the root folder
+        auto alias = _model->data(index, FolderStatusDelegate::FolderAliasRole).toString();
+        if (Folder *f = FolderMan::instance()->folder(alias)) {
+            folderUrl = QUrl::fromLocalFile(f->path());
+        }
+    }
+
+    if (!folderUrl.isEmpty()) {
+        QString openLocallyLabel;
+        if (Utility::isWindows()) {
+            openLocallyLabel = tr("Show in Explorer");
+        } else if (Utility::isMac()) {
+            openLocallyLabel = tr("Show in Finder");
+        } else {
+            openLocallyLabel = tr("Show if file manager");
+        }
+
+        QAction *ac = menu->addAction(openLocallyLabel);
+        connect(ac, &QAction::triggered, [folderUrl]() {
+            qCInfo(lcAccountSettings) << "Opening local folder" << folderUrl;
+            if (!QDesktopServices::openUrl(folderUrl)) {
+                qCWarning(lcAccountSettings) << "QDesktopServices::openUrl failed for" << folderUrl;
+            }
+        });
+
+        if (!QFile::exists(folderUrl.toLocalFile())) {
             ac->setEnabled(false);
         }
+    }
+
+    // Add an action to open the folder on the server in a webbrowser:
 
-        ac = menu->addAction(tr("Open folder in browser"));
-        auto info = _model->infoForIndex(index);
-        OC_ASSERT(info);
+    if (auto info = _model->infoForIndex(index)) {
+        QAction *ac = menu->addAction(tr("Open Remote Folder in Browser"));
         QString path = info->_folder->remotePathTrailingSlash();
         path += info->_path;
```

Comment:
for the root folder this results in //

Reply:
No, `remotePathTrailingSlash()` only appends the slash if there isn't already a trailing slash.

Reply:
yes but info->_path will be `/`

Reply:
I've tried :D